scipy/special/_amos.c

```diff
@@ -0,0 +1,6401 @@
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "_amos.h"
+
+#ifndef CMPLX
+#define CMPLX(x, y) ((double complex)((double)(x) + I * (double)(y)))
+#endif /* CMPLX */
+
+static int amos_acai(double complex, double, int, int, int, double complex *, double, double, double, double);
+static int amos_acon(double complex, double, int, int, int, double complex *, double, double, double, double, double);
+static int amos_asyi(double complex, double, int, int, double complex *, double, double, double, double);
+static int amos_binu(double complex, double fnu, int, int, double complex *, double, double, double, double, double);
+static int amos_bknu(double complex, double, int, int, double complex *, double, double, double);
+static int amos_buni(double complex, double, int, int, double complex *, int, int *, double, double, double, double);
+static int amos_bunk(double complex, double, int, int, int, double complex *, double, double, double);
+static double amos_gamln(double);
+static int amos_kscl(double complex, double, int, double complex *, double complex, double *, double, double);
+static int amos_mlri(double complex, double, int, int, double complex *, double);
+static void amos_rati(double complex, double, int, double complex *, double);
+static int amos_seri(double complex, double, int, int, double complex *, double, double, double);
+static int amos_s1s2(double complex, double complex *, double complex *, double, double, int *);
+static int amos_uchk(double complex, double, double);
+static void amos_unhj(double complex, double, int, double, double complex *, double complex *, double complex *, double complex *, double complex *, double complex *);
+static void amos_uni1(double complex, double, int, int, double complex *, int *, int *, double, double, double, double);
+static void amos_uni2(double complex, double, int, int, double complex *, int *, int *, double, double, double, double);
+static void amos_unik(double complex, double, int, int, double, int *, double complex *, double complex *, double complex *, double complex *, double complex *);
+static int amos_unk1(double complex, double, int, int, int, double complex *, double, double, double);
+static int amos_unk2(double complex, double, int, int, int, double complex *, double, double, double);
+static int amos_uoik(double complex, double, int, int, int, double complex *, double, double, double);
+static int amos_wrsk(double complex, double, int, int, double complex *, double complex *, double, double, double);
```

Comment:
Should these be in the header file instead?

Reply:
Not really a C veteran but if I understand correctly static keeps it local to this file and does not pollute the namespace of the library. Or at least that's my limited understanding of things you wish to keep local. So these are the prototypes for such elements.